pages/implementation/reference/javascript.mdx

```diff
@@ -0,0 +1,201 @@
+---
+layout: 'guide'
+title: 'JavaScript'
+---
+
+import Link from '../../../components/Link';
+
+### Avo can generate JavaScript code based on your tracking plan.
+
+The code gives you access to JavaScript functions, each function representing an event, identification, user properties update, page view, revenue or mix of those.
+
+We call those functions Avo functions.
+
+You invoke the Avo functions, provide tracking parameters and 
+- Avo takes care of verifying that your tracking call is correct according to the defined tracking plan
+- Avo also tracks your functions usage in development and reports if a call of particular function was seen or not. 
+
+> It's useful to get the overview of the tracking implementartion when working on new tracking. 
+You can also get more insights into dev implementation and actual production state of your tracking with <Link passHref href="/workspace/tracking-plan/implementation-status#inspector-implementation-status"><a>the Inspector implementation status</a></Link>
+
+- Eventually Avo sends the data to the destinations it manages and calls the callbacks (called custom destinations) you provided when initializing
+
+### Handling tracking plan changes
+
+When tracking plan changes, you generate the file again and swap the old file with the new one. Avo manages the merge conflicts inside the file.
+Then, if the existing events changed, you'll have to update your Avo functions calls according to the new tracking plan to pass the runtime checks in the Avo file.
+You'll also get access to the new events defined in your tracking plan after the previous generated code update.
+
```

Comment:
I feel like everything up to this point could be a more generic intro to Avo Functions in the docs. I think the Reference doc should be purely focused on the technical specs, and how to get up and running.

Reply:
But will people read the generic doc before this one? I wanted this to have enough info to get started with some basic understanding

Reply:
We should reference it on top to point people towards it if they feel the need to. I feel like the expectation when opening a specific reference doc is "just show me how it works", and skip the focus on value props etc, that's handled in the guides.

Reply:
I see, was not treating this as a 100% reference page

Reply:
Ye I think it's good to have some text around things, but I think we'll have other sections focusing on the Problem & Solution part